File: web/src/type/password.ts
// COMMON (WEB AND SERVER)

export interface Password {
    username: string,
    password: string,
    type: PasswordType,
}

export enum PasswordType {
    POSTGRES,
    PATRONI,
}

export interface PasswordMap {
    [uuid: string]: Password,
}

// SPECIFIC (WEB)

File: service/src/model/password.go
package model

// COMMON (WEB AND SERVER)

type Password struct {
	Username string       `json:"username"`
	Password string       `json:"password"`
	Type     PasswordType `json:"type"`
}

type PasswordType int

const (
	POSTGRES PasswordType = iota
	PATRONI
)

type PasswordMap map[string]Password

// SPECIFIC (SERVER)
